spark/wordcount/src/main/scala/com/example/bigtable/spark/wordcount/WordCount.scala

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigtable.spark.wordcount
+
+import com.google.cloud.bigtable.hbase.BigtableConfiguration
+import org.apache.hadoop.hbase.{HColumnDescriptor, HConstants, HTableDescriptor, TableName}
+import org.apache.hadoop.hbase.mapreduce.{TableInputFormat, TableOutputFormat}
+import org.apache.hadoop.hbase.client.{Connection, Put}
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable
+import org.apache.hadoop.hbase.util.Bytes
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.SparkContext
+
+/**
+  * Basic WordCount sample of using Cloud Dataproc (managed Apache Spark)
+  * to write to Cloud Bigtable.
+  */
+object WordCount {
+  val ColumnFamily = "cf"
+  val ColumnFamilyBytes = Bytes.toBytes(ColumnFamily)
+  val ColumnNameBytes = Bytes.toBytes("Count")
+
+  def createConnection(ProjectId: String, InstanceID: String): Connection = {
+    BigtableConfiguration.connect(ProjectId, InstanceID)
+  }
+
+  /**
+    * Create a table in the Cloud Bigtable instance if it doesn't already
+    * exit.
+    *
+    * @param connection A Cloud Bigtable connection
+    * @param name       The table name
+    */
+  def createTableIfNotExists(connection: Connection, name: String) = {
+    val tableName = TableName.valueOf(name)
+    val admin = connection.getAdmin()
+    try {
+      if (!admin.tableExists(tableName)) {
+        val tableDescriptor = new HTableDescriptor(tableName)
+        tableDescriptor.addFamily(
+          new HColumnDescriptor(ColumnFamily))
+        admin.createTable(tableDescriptor)
+      }
+    } finally {
+      admin.close()
+    }
+  }
+
+  /**
+    * Main entry point for running the WordCount spark job and writing
+    * the results to Bigtable.
+    *
+    * @param projectId  The Google Cloud Project ID
+    * @param instanceId The Cloud Bigtable instance name
+    * @param tableName  The name of the Cloud Bigtable table to write to
+    * @param fileName   The file (local or gcs) to sort
+    * @param sc         The Spark context
+    */
+  def runner(projectId: String,
+             instanceId: String,
+             tableName: String,
+             fileName: String,
+             sc: SparkContext) = {
+    val createTableConnection = createConnection(projectId, instanceId)
+    try {
+      createTableIfNotExists(createTableConnection, tableName)
+    } finally {
+      createTableConnection.close()
+    }
+
+    var conf = BigtableConfiguration.configure(
+      projectId, instanceId)
+    conf.set(TableInputFormat.INPUT_TABLE, tableName)
+    conf.set(TableOutputFormat.OUTPUT_TABLE, tableName)
+    conf.setInt(HConstants.HBASE_CLIENT_OPERATION_TIMEOUT, 60000)
+    conf.set(
+      "hbase.client.connection.impl",
+      "com.google.cloud.bigtable.hbase1_x.BigtableConnection")
```

Comment:
Please remove this:

>    conf.set(
>      "hbase.client.connection.impl",
>      "com.google.cloud.bigtable.hbase1_x.BigtableConnection")

BigtableConfiguration.configure() already does the right thing, and will be more resilient overtime as we change the Connection class (like in hbase2).